OpenRA.Mods.Common/Traits/AutoTarget.cs

```diff
@@ -53,6 +53,7 @@ public enum UnitStance { HoldFire, ReturnFire, Defend, AttackAnything }
 	public class AutoTarget : ConditionalTrait<AutoTargetInfo>, INotifyIdle, INotifyDamage, ITick, IResolveOrder, ISync
 	{
 		readonly AttackBase[] attackBases;
+		readonly IEnumerable<AttackBase> activeAttackBases;
```

Comment:
I'm not worried about the speed here - just avoiding the repeated allocation. Since it has 4 uses it would need 4 explicit loops which is not worth it readability wise.

For attackFollows, I can't cache the delegate since it captures the `allowMove` parameter. An explicit loop is the most reasonable way of avoiding the repeated allocations in this case.